### PR TITLE
Nessie Server Admin Tool: add support for MariaDB and MySQL backends

### PR DESCRIPTION
**Pull request by adutra**
Requires #8544

### CODE REVIEW

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/AbstractContentInfoTests.java
Comment (adutra) [inline tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/AbstractContentInfoTests.java:36]: These tests weren't being executed (the class is abstract).

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/ITCheckContentPersist.java
Comment (adutra) [inline tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/ITCheckContentPersist.java:34]: Refactored to remove the nested test, as nested tests do not work well with Quarkus tests.

File: tools/server-admin/src/main/java/org/projectnessie/tools/admin/cli/CheckContent.java
Comment (adutra) [inline tools/server-admin/src/main/java/org/projectnessie/tools/admin/cli/CheckContent.java:247]: Removed, since now there is only one impl left.

File: tools/server-admin/build.gradle.kts
Comment (snazy) [inline tools/server-admin/build.gradle.kts:157]: Better reduce to `1` (safer ;)

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestBackends.java
Comment (snazy) [inline tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestBackends.java:30]: I know that @dimas-b will complain about lower case enum value ;) ;)

File: tools/server-admin/build.gradle.kts
Comment (adutra) [reply tools/server-admin/build.gradle.kts:157]: With 1, the overall build becomes quite slow because a new container is started for every combination of test class + backend. With 4, a new container is spawned just when the backend changes.

**Review by snazy — COMMENTED** [review]
Overall LGTM, left a couple comments

File: tools/server-admin/build.gradle.kts
Comment (snazy) [reply tools/server-admin/build.gradle.kts:157]: Okay - then let's go with 4 for now.

However, I suspect there's a classloader/resource leak related to tests.

Comment (snazy) [reply tools/server-admin/build.gradle.kts:157]: > However, I suspect there's a classloader/resource leak related to tests.

Can be investigated separately.

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java
Comment (snazy) [inline tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java:161]: Fun - GitHub doesn't recognize the `record` keyword :)

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java
Comment (snazy) [inline tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java:97]: Why running GC?

**Comment by snazy** [comment]
> Looks like there's a resource leak. I see this locally when running `./gradlew :nessie-server-admin-tool:intTest`

NVM - that's fixed via #8549

**Review by snazy — COMMENTED** [review]
Overall LGTM - but needs a rebase, once the other PRs are merged.

**Comment by adutra** [comment]
> > Looks like there's a resource leak. I see this locally when running `./gradlew :nessie-server-admin-tool:intTest`
> 
> NVM - that's fixed via #8549

Indeed. But I suspect there is another leak somewhere. I fear it could be a consequence of mixing together `QuarkusMainTest` and `MultiEnvTestEngine` (it's the first time we try).

File: tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java
Comment (adutra) [reply tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java:97]: Whoops leftover, I will remove. I tried a few desperate things that didn't work (of course).

File: tools/server-admin/build.gradle.kts
Comment (adutra) [reply tools/server-admin/build.gradle.kts:157]: https://github.com/projectnessie/nessie/issues/8571

**Review by snazy — APPROVED** [review]

**Pull request merged**
